Objects/frameobject.c

```diff
@@ -310,6 +310,13 @@ framelocalsproxy_dealloc(PyObject *self)
 static PyObject *
 framelocalsproxy_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
 {
+    if (PyTuple_GET_SIZE(args) != 1) {
+        PyErr_Format(PyExc_TypeError,
+                     "FrameLocalsProxy expected 1 argument, got %zd",
+                     PyTuple_GET_SIZE(args));
+        return NULL;
+    }
+
```

Comment:
Should we also raise a TypeError if the object in the tuple is not a frame? Looks like that would currently trigger an assertion below.

Reply:
Right, I updated my PR to check that the argument is actually a frame.